tests/suite/system_suite_test.go

```diff
@@ -101,38 +130,84 @@ var _ = BeforeSuite(func() {
 		version = "edge"
 	}
 
-	clusterInfo, err = resourceManager.GetClusterInfo()
-	Expect(err).ToNot(HaveOccurred())
-
 	output, err := framework.InstallGatewayAPI(k8sClient, *gatewayAPIVersion, *k8sVersion)
 	Expect(err).ToNot(HaveOccurred(), string(output))
 
-	output, err = framework.InstallNGF(cfg)
+	output, err = framework.InstallNGF(installCfg, extraInstallArgs...)
 	Expect(err).ToNot(HaveOccurred(), string(output))
 
-	podName, err := framework.GetNGFPodName(k8sClient, cfg.Namespace, cfg.ReleaseName, timeoutConfig.CreateTimeout)
+	podNames, err := framework.GetNGFPodNames(
+		k8sClient,
+		installCfg.Namespace,
+		installCfg.ReleaseName,
+		timeoutConfig.CreateTimeout,
+	)
 	Expect(err).ToNot(HaveOccurred())
 
 	if *serviceType != "LoadBalancer" {
-		portFwdPort, err = framework.PortForward(k8sConfig, cfg.Namespace, podName, portForwardStopCh)
+		portFwdPort, err = framework.PortForward(k8sConfig, installCfg.Namespace, podNames[0], portForwardStopCh)
```

Comment:
I think it is possible for `podNames` to be nil here if all NGF Pods are not ready.